c_src/sasl_auth.c

```diff
@@ -243,6 +245,116 @@ static ERL_NIF_TERM sasl_cli_step(ErlNifEnv* env, int UNUSED(argc), const ERL_NI
     return ret;
 }
 
+// server begin
+static ERL_NIF_TERM sasl_srv_new(ErlNifEnv* env, int UNUSED(argc), const ERL_NIF_TERM argv[])
+{
+    ErlNifBinary service, principal;
+    sasl_state_t* state = NULL;
```

Comment:
nit: Setting state to NULL seems unnecessary here. I would prefer if state and result are declared where they are initialized (C99 and later supports this).